### PR TITLE
Mcux lpuart async fixes

### LINKED ISSUE #79323: tests: uart: uart_async_api: test failure on mimxrt1170/1160 /1180
**Describe the bug**

uart_asunc_api test for drivers.uart.async_api.lpuart fails

Please also mention any information which could help others to understand
the problem you're facing:
 - What target platform are you using?
 - mimxrt1170_evk/mimxrt1160_evk/mimxrt1180_evk
 - What have you tried to diagnose or workaround this issue?
 - NA
 - Is this a regression? If yes, have you been able to "git bisect" it to a
   specific commit?
 - No, this has been there since the test first introduced. and which is blocked and can only be tested after #78023 merged

**To Reproduce**

1. west build -b mimxrt1170_evk/mimxrt1176/cm7 tests/drivers/uart/uart_async_api/ -- -DCONFIG_DCACHE=y -DCONFIG_NOCACHE_MEMORY=y -DCONFIG_USERSPACE=n
2. west flash
3. see error

**Expected behavior**

Test PASS

**Impact**

uart async functions

**Logs and console output**

```
*** Booting Zephyr OS build v3.7.0-3802-g9655c2f61b8b ***
Running TESTSUITE uart_async_chain_read
===================================================================
E: No buffers to release from RX DMA!
E: No buffers to release from RX DMA!
START - test_chained_read
E: Got error : -5

    Assertion failed at WEST_TOPDIR/zephyr/tests/drivers/uart/uart_async_api/src/test_uart_async.c:348: test_chained_read_callback: rx_data_idx + evt->data.e

ASSERTION FAIL [0] @ WEST_TOPDIR/zephyr/kernel/sched.c:1430
        aborting essential thread 0x80002570
E: r0/a1:  0x00000004  r1/a2:  0x00000596  r2/a3:  0x4007c000
E: r3/a4:  0x00000004 r12/ip:  0x00000000 r14/lr:  0x3000a0a3
E:  xpsr:  0x21000000
E: Faulting instruction address (r15/pc): 0x3000b3a0
E: >>> ZEPHYR FATAL ERROR 4: Kernel panic on CPU 0
E: Current thread: 0x80002570 (sysworkq)
E: Halting system
```

**Environment (please complete the following information):**

 - OS: (e.g. Linux, MacOS, Windows)
 - Toolchain (e.g Zephyr SDK, ...)
 - Commit SHA or Version used: v3.7.0-3802-g9655c2f61b8b

### PR DESCRIPTION
**Pull request by hakehuang**
uart_async_api for NXP RT11xx series async testing failure, and after check the uart_mcux_lpuart.c, there are several issues found.
and the uart_async_api test need also some change to fit in DMA context.

fixes: #79323

### CODE REVIEW

**Comment by hakehuang** [comment]
@EmilioCBen  please help to review, this is a hidden issue after your fixes. Thanks

File: drivers/serial/uart_mcux_lpuart.c [hunk 1]
```diff
@@ -584,7 +585,7 @@ static void prepare_rx_dma_block_config(const struct device *dev)
 	head_block_config->dest_address = (uint32_t)rx_dma_params->buf;
 	head_block_config->source_address = LPUART_GetDataRegisterAddress(lpuart);
 	head_block_config->block_size = rx_dma_params->buf_len;
-	head_block_config->dest_scatter_en = true;
+	head_block_config->dest_scatter_en = false;
```
[the remark this answers is not in the document]
Comment (hakehuang) [reply drivers/serial/uart_mcux_lpuart.c:588]: yes, if we set the scatter_en to true, dam_reload may have problem. as the uart driver does not realy use the scatter method. and enable scatter without update the list will cause problem, and the dma get no data.

File: drivers/serial/uart_mcux_lpuart.c [hunk 2]
```diff
@@ -615,20 +616,38 @@ static int uart_mcux_lpuart_dma_replace_rx_buffer(const struct device *dev)
...
 	/* There must be a buffer to replace this one with */
 	assert(data->async.next_rx_buffer != NULL);
 	assert(data->async.next_rx_buffer_len != 0U);
-	const int success = dma_reload(config->rx_dma_config.dma_dev,
-				       config->rx_dma_config.dma_channel,
-				       LPUART_GetDataRegisterAddress(lpuart),
-				       (uint32_t)data->async.next_rx_buffer,
-				       data->async.next_rx_buffer_len);
+	rx_dma_params->buf = data->async.next_rx_buffer;
+	rx_dma_params->buf_len = data->async.next_rx_buffer_len;
```
[the remark this answers is not in the document]
Comment (hakehuang) [reply drivers/serial/uart_mcux_lpuart.c:627]: no, as we do not use the scatter way in reload, which requires to link the buffer when we call the dma-config, so hardware does not know this. as in async mode, the next dma buffer is assigned by user application when it is required, so we can not pre-create the scatter list.

**Review by dleach02 — APPROVED** [review]

File: tests/drivers/uart/uart_async_api/src/test_uart_async.c [hunk 3]
```diff
@@ -324,17 +324,24 @@ ZTEST_USER(uart_async_multi_rx, test_multiple_rx_enable)
 }
 
 #if NOCACHE_MEM
-static __aligned(32) uint8_t chained_read_buf[2][8] __used __NOCACHE;
+/* To ensure 32-bit alignment of the buffer array,
+ * the two arrays are defined instead using an array of arrays
+ */
+static __aligned(32) uint8_t chained_read_buf_0[8] __used __NOCACHE;
```
Comment (JarmouniA) [inline tests/drivers/uart/uart_async_api/src/test_uart_async.c:330]: I have a question that's unrelated to your changes, does `__used` have any utility here?

Comment (hakehuang) [reply tests/drivers/uart/uart_async_api/src/test_uart_async.c:330]: @JarmouniA ,  __used is to force compiler to emit symbol, but I do not know whether it is a must in Zephyr, so I copy the line headless.

**Review by EmilioCBen — APPROVED** [review]

**Pull request merged**